.github/workflows/nightly_release.yaml

```diff
@@ -37,9 +37,30 @@ jobs:
     runs-on: ubuntu-22.04
     steps:
       - name: Harden Runner
-        uses: step-security/harden-runner@63c24ba6bd7ba022e95695ff85de572c04a18142 # v2.7.0
+        uses: step-security/harden-runner@17d0e2bd7d51742c71671bd19fa12bdc9d40a3d6 # v2.8.1
         with:
-          egress-policy: audit
+          egress-policy: block
+          # When adding endpoints, see README.md.
+          # prettier-ignore
+          allowed-endpoints: >
+            api.github.com:443
+            bcr.bazel.build:443
+            cfhcable.dl.sourceforge.net:443
+            cytranet-dal.dl.sourceforge.net:443
+            downloads.sourceforge.net:443
+            gigenet.dl.sourceforge.net:443
+            github.com:443
+            netactuate.dl.sourceforge.net:443
+            oauth2.googleapis.com:443
+            objects.githubusercontent.com:443
+            phoenixnap.dl.sourceforge.net:443
+            pilotfiber.dl.sourceforge.net:443
+            psychz.dl.sourceforge.net:443
+            releases.bazel.build:443
+            sourceforge.net:443
+            storage.googleapis.com:443
+            uploads.github.com:443
+            versaweb.dl.sourceforge.net:443
```

Comment:
I don't know how to audit these `*.dl.sourceforge.net` domains. Can we blanket permit anything of the form `*.dl.sourceforge.net:443`? (If not, how do we know which ones are safe?)

Reply:
Seems to have worked, that's the best I can say because the syntax is badly undocumented. pre-commit uses the in-progress workflow file.

I'd also tried `|` instead of `>` to workaround formatting, but that didn't work.